bokehjs/src/lib/models/tools/edit/line_edit_tool.ts

```diff
@@ -48,8 +48,24 @@ export class LineEditToolView extends LineToolView {
     }
 
     const cds = this._selected_renderer.data_source
-    const x = Array.from(cds.data.x)
-    const y = Array.from(cds.data.y)
+
+    const glyph: any = this._selected_renderer.glyph
+    const [xkey, ykey] = [glyph.x.field, glyph.y.field]
+
+    let x: number[]
+    let y: number[]
+
+    if (xkey) {
+      x = cds.get_array(xkey)
+    } else {
+      x = glyph.x.value
```

Comment:
Apologies @scaine1  in your question I overlooked the `.value` part and answered a different question. The `.value` vs `.field` distinguishes between the *x* and *y* usage here:
```
plot.circle(x=5, y="foo")
```
i.e. `glyph.x.value` represents when a glyph coordinate is configured with a fixed value like `x=5`. But these tools can't work in that case, they have to have a CDS column to operate on, so there is no reason to try to handle that case. i.e. the "else" blocks on both of these can just be removed